docker/build-tools/build-and-push.sh

```diff
@@ -16,9 +16,18 @@
 
 # support other container tools, e.g. podman
 CONTAINER_CLI=${CONTAINER_CLI:-docker}
-HUB=gcr.io/istio-testing
-VERSION=$(date +%Y-%m-%dT%H-%M-%S)
 
-${CONTAINER_CLI} build -t "${HUB}/build-tools:${VERSION}" -t "${HUB}/build-tools:latest" .
+HUB=${HUB:-gcr.io/istio-testing}
+DATE=$(date +%Y-%m-%dT%H-%M-%S)
+BRANCH=$(git branch --show-current 2>/dev/null || echo "master")
+VERSION="${BRANCH}-${DATE}"
+
+${CONTAINER_CLI} build -t "${HUB}/build-tools:${VERSION}" -t "${HUB}/build-tools:${BRANCH}-latest" .
+
 ${CONTAINER_CLI} push "${HUB}/build-tools:${VERSION}"
-${CONTAINER_CLI} push "${HUB}/build-tools:latest"
+${CONTAINER_CLI} push "${HUB}/build-tools:${BRANCH}-latest"
+
+if [ "${BRANCH}" == "master" ]; then
```

Comment:
Do you also want a ':latest' in addition to `:master-latest'?

Reply:
I don't think we strictly need it. It's useful if someone wants to refer to `latest`, and that will always mean the latest build from master, not the last succeeded build (as it is today).

If you feel we don't want this I can remove this section.

Reply:
Well, `:master-latest` would be the same as `:latest` so I don't think you need this section. But if someone is using `:latest` they will stuck with the build before this one.

Reply:
Yeah, which leads the question: Is someone actually using `latest`? Can we delete that tag?

Reply:
tag removed.